### PR TITLE
APPCLI-56: Added support for setting encrypted values directly

### PR DESCRIPTION
**Pull request by adamkimpton**
Made the `value` argument from the `app configure set` command and the `text` argument from the `app encrypt` command optional, this is to avoid having passwords appear in the user's terminal history.
Created an `--encrypted` flag for the `app configure set` command to encrypt the value before setting it.
Moved functionality from the encrypt function into a helper functions file to have access to it from `encrypt_cli.py` & `configure_cli.py` as I figured this was easier than using click's [invoke](https://click.palletsprojects.com/en/5.x/advanced/#invoking-other-commands) function.
Resolves: #38

### CODE REVIEW

File: appcli/commands/configure_cli.py [hunk 1]
```diff
@@ -145,21 +146,34 @@ def get(ctx, setting):
...
+            "-e",
+            "--encrypted",
+            is_flag=True,
+        )
         @click.argument("setting")
-        @click.argument("value")
+        @click.argument("value", required=False)
         @click.pass_context
-        def set(ctx, type, setting, value):
+        def set(ctx, type, encrypted, setting, value):
```
[the remark this answers is not in the document]
Comment (adamkimpton) [reply appcli/commands/configure_cli.py:158]: Added python doc block, might need additionial changes

File: appcli/functions.py [hunk 2]
```diff
@@ -51,6 +54,25 @@ def print_header(title):
         title.upper(),
     )
 
+def encrypt_text(cli_context, text):
```
Comment (adamkimpton) [inline appcli/functions.py:57]: Moved `encrypt_helper` function from `appcli/appcli/commands/helpers.py` to `appcli/functions.py` and renamed function to `encrypt_text`

**Comment by thomas-anderson-bsl** [comment]
Hey @adamkimpton  I've finished review of 'setting encrypted values directly' and have made a commit just making some minor adjustments
https://github.com/brightsparklabs/appcli/pull/82/commits/dfed299467ef9977cbdefad49646ecbc57001c30
Check it out and let me know if you have any questions. Of note:
- added typing to function arguments where possible (without creating circular references and dealing with those...)
- handled the case in configure set where we cannot support encrypting a non-string type value
- the prompts weren't quite right, click adds the ':' to the end of the string you provide to click.prompt
- exit early on encrypt_text function if the text is None. In general, you should look to exit functions as early as possible with erroneous cases.

**Comment by thomas-anderson-bsl** [comment]
CI - the first set of unit tests pass, the only failure is the docker publish.
CI is being weird about `docker login` even though credentials should be present.
I'm going to merge this branch regardless and deal with CI separately.

**Pull request merged**